python/cuml/cluster/kmeans.pyx

```diff
@@ -281,6 +281,12 @@ class KMeans(Base):
         cdef KMeansParams params
         params.n_clusters = <int>self.n_clusters
 
+        # K-means++ is the constrained case of k-means||
+        # w/ oversampling factor = 0
+        if (init == 'k-means++'):
+            init = 'k-means||'
+            self.oversampling_factor = 0
```

Comment:
Quick questions: If the user accidentally provides `init="k-means++"` and also passes an oversampling_factor!=0, this value gets overwritten by the param below. I'm not too familiar with the algorithm but if `k-means++` is a constrained case does it make sense to add some sort of warning here telling the user that both have been passed?

Reply:
I was just planning to document the oversampling factor as being ignored when `init = 'kmeans++'` as I see scikit-learn often doing for these types of cases. I suppose issuing a warning in addition to the documentation can also help clarify this for users who haven't looked at the docs.

Reply:
Agree that we need to run multiple times. Serial will get different answers, and multi-init (parallel kmean|| init) is different than that even.  One key thing to remember, unless we have a 'blobs' like data set, with clearly separated clusters, (the one above isn't one of those) there will always be multiple answers, which are all mostly correct.  Only running multiple times or adding in another criteria (like maximizing the distance between centroids) can get a unique answer.

Reply:
> One key thing to remember, unless we have a 'blobs' like data set, with clearly separated clusters, (the one above isn't one of those) there will always be multiple answers,

I totally agree and understand this point. My concern in my above example is not that I was getting different answers, but rather I was getting different results for the exact same random_state setting, which tells me something is not using the seed that's being passed in. While I'd still expect the centroids to have very small round-off errors in that case from parallel arithmetic, I'd still expect consistency in the resulting labels across runs with the same random_state.

Reply:
That being said, I believe that specific issue has been resolved now because I'm no longer getting this inconsistency. The resulting inertia is also sometimes even better than sklearn now. This doesn't seem to be the case without the `n_init` (basically for some seeds it's just very bad).

Reply:
That being said, the variance issue I was having was related to the generated dataset being shuffled inside of `make_blobs` before it was returned. Unfortunately, the shuffle wasn't honoring the provided random seed so it was producing different results. The resulting inertia is also sometimes even better than sklearn now. This doesn't seem to be the case without the `n_init` (basically for some seeds it's just very bad).